src/dash-table/components/EdgeFactory.tsx

```diff
@@ -0,0 +1,314 @@
+            style_header,
+            style_header_conditional,
+            virtualized.data,
+            virtualized.offset
+        );
+    }
+
+    private memoizedCreateEdges = memoizeOne((
+        active_cell: ICellCoordinates,
+        columns: VisibleColumns,
+        operations: number,
+        filtering: boolean,
+        filterMap: Map<string, SingleColumnSyntaxTree>,
+        _n_fixed_columns: number,
+        n_fixed_rows: number,
+        style_as_list_view: boolean,
+        style_cell: Style,
+        style_cell_conditional: Cells,
+        style_data: Style,
+        style_data_conditional: DataCells,
+        style_filter: Style,
+        style_filter_conditional: BasicFilters,
+        style_header: Style,
+        style_header_conditional: Headers,
+        data: Data,
+        offset: IViewportOffset
+    ) => {
+        const dataStyles = this.dataStyles(
+            style_cell,
+            style_data,
+            style_cell_conditional,
+            style_data_conditional
+        );
+
+        const filterStyles = this.filterStyles(
+            style_cell,
+            style_filter,
+            style_cell_conditional,
+            style_filter_conditional
+        );
+
+        const headerStyles = this.headerStyles(
+            style_cell,
+            style_header,
+            style_cell_conditional,
+            style_header_conditional
+        );
+
+        const headerRows = getHeaderRows(columns);
+
+        let dataEdges = this.getDataEdges(
+            columns,
+            dataStyles,
+            data,
+            offset,
+            active_cell,
+            style_as_list_view
+        );
+
+        let dataOpEdges = this.getDataOpEdges(
+            operations,
+            dataStyles,
+            data,
+            offset,
+            style_as_list_view
+        );
+
+        let filterEdges = this.getFilterEdges(
+            columns,
+            filtering,
+            filterMap,
+            filterStyles,
+            style_as_list_view
+        );
+
+        let filterOpEdges = this.getFilterOpEdges(
+            operations,
+            filtering,
+            filterStyles,
+            style_as_list_view
+        );
+
+        let headerEdges = this.getHeaderEdges(
+            columns,
+            headerRows,
+            headerStyles,
+            style_as_list_view
+        );
+
+        let headerOpEdges = this.getHeaderOpEdges(
+            operations,
+            headerRows,
+            headerStyles,
+            style_as_list_view
+        );
+
+        const cutoffWeight = (style_cell ? 1 : 0) + style_cell_conditional.length - 1;
+
+        headerEdges = EdgeFactory.clone(headerEdges);
+        headerOpEdges = EdgeFactory.clone(headerOpEdges);
+        filterEdges = EdgeFactory.clone(filterEdges);
+        filterOpEdges = EdgeFactory.clone(filterOpEdges);
+        dataEdges = EdgeFactory.clone(dataEdges);
+        dataOpEdges = EdgeFactory.clone(dataOpEdges);
```

Comment:
Clone everything before operating on it... we want to keep the previous memoized results clean